sdk/tables/Azure.Data.Tables/src/TableOdataFilter.cs

```diff
@@ -80,5 +80,9 @@ public static string Create(FormattableString filter)
 
             return string.Format(CultureInfo.InvariantCulture, filter.Format, args);
         }
+
+        internal static string EscapeStringValue(string s) => s.Replace("'", "''");
+        internal static StringBuilder EscapeStringValue(StringBuilder s) => s.Replace("'", "''");
+        internal static string EscapeStringValue(char s) => s.ToString().Replace("'", "''");
```

Comment:
Should we consider short-circuiting this if `s` is not the single quote character?   Maybe something like:

```csharp
internal static string EscapeStringValue(char s) => 
    s switch
    {
        _ when s == '\'' => "''",
        _ => s.ToString()
    };
```